src/handlers/motions/motionStaked/helpers.ts

```diff
@@ -0,0 +1,209 @@
+    stakes: {
+      raw: {
+        ...existingUserStakes.stakes.raw,
+        [stakedSide]: updatedRawStake.toString(),
+      },
+      percentage: {
+        ...existingUserStakes.stakes.percentage,
+        [stakedSide]: updatedPercentage.toString(),
+      },
+    },
+  };
+
+  return updatedUserStakes;
+};
+
+type UserStakesMapFn = (usersStakes: UserStakes) => UserStakes;
+
+const getUserStakesMapFn =
+  (
+    staker: string,
+    vote: BigNumber,
+    amount: BigNumber,
+    requiredStake: BigNumber,
+  ): UserStakesMapFn =>
+  (userStakes: UserStakes): UserStakes => {
+    const { address } = userStakes;
+
+    if (address === staker) {
+      return getUpdatedUserStakes(userStakes, amount, vote, requiredStake);
+    }
+
+    return userStakes;
+  };
+
+/**
+ * Takes existing usersStakes and updates the entry with the latest stake data
+ */
+export const getUpdatedUsersStakes = (
+  usersStakes: UserStakes[],
+  staker: string,
+  vote: BigNumber,
+  amount: BigNumber,
+  requiredStake: BigNumber,
+): UserStakes[] => {
+  const isExistingStaker = usersStakes.some(
+    ({ address }) => address === staker,
+  );
+
+  if (isExistingStaker) {
+    const updateUserStakes = getUserStakesMapFn(
+      staker,
+      vote,
+      amount,
+      requiredStake,
+    );
+
+    return usersStakes.map(updateUserStakes);
+  }
+
+  const newUserStakes = getNewUserStakes(staker, vote, amount, requiredStake);
+
+  return [...usersStakes, newUserStakes];
+};
+
+export const getUserMinStake = (
+  totalStakeFraction: BigNumber,
+  userMinStakeFraction: BigNumber,
+  skillRep: BigNumber,
+): string => {
+  /*
+   * Both totalStakeFraction and userMinStakeFraction both divide by 10 to the power of 18.
+   * Since we've multiplied by both, we need to divide by 10 to the power of 18, twice.
+   */
+
+  return skillRep
+    .mul(totalStakeFraction)
+    .mul(userMinStakeFraction)
+    .div(BigNumber.from(10).pow(36))
+    .toString();
+};
+
+/*
+ * Motion ids are not unique between voting reputation installations. If you had a motion with id of 1
+ * in a previous voting reputation installation, you will get another one after uninstalling and reinstalling
+ * the voting reputation client. So, we only want the one created most recently, since that's the one created
+ * by the currently installed version of the voting reputation extension.
+ */
+
+export const getStakedMotion = (
+  motions: MotionQuery[],
+  motionId: BigNumber,
+): MotionQuery | undefined => {
+  const motionsWithCorrectId = motions.filter(
+    ({ motionData: { motionId: id } }) => id === motionId.toString(),
+  );
+
+  if (motionsWithCorrectId.length > 1) {
+    motionsWithCorrectId.sort(
+      (a, b) =>
+        new Date(a.createdAt).valueOf() - new Date(b.createdAt).valueOf(),
+    );
+  }
+
+  const stakedMotion = motionsWithCorrectId.pop();
+
+  if (!stakedMotion) {
+    verbose(
+      'Could not find the staked motion in the db. This is a bug and needs investigating.',
+    );
```

Comment:
We should always be able to find the motion that is being staked in the db (because it gets put there when it gets created). 

So this should never happen. Leaving a log for debugging purposes in the event it does.